public/components/trace_analytics/components/services/service_view.tsx

```diff
@@ -76,6 +77,18 @@ export function ServiceView(props: ServiceViewProps) {
   const [redirect, setRedirect] = useState(false);
   const [actionsMenuPopover, setActionsMenuPopover] = useState(false);
 
+  let serviceId: string | null = null;
+
+  try {
+    const location = useLocation();
+    const params = new URLSearchParams(location?.search || '');
+    serviceId = params.get('serviceId');
+  } catch (error) {
+    serviceId = null;
+  }
```

Comment:
Can we use hooks and state here to make it cleaner?

```
  const location = useLocation();

  useEffect(() => {
    try {
      const params = new URLSearchParams(location.search);
      const id = params.get('serviceId');
      setServiceId(id);
    } catch (error) {
      setServiceId(null);
    }
  }, [location]);
  ```

Reply:
Done. Thanks for the feedback =)